doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -161,6 +161,11 @@ to accessible objects.  (see Section :ref:`invocation-of-tactics`).
         :n:`@smart_qualid`).  Cf. :ref:`this example
         <search-by-notation>`.
 
+   * :n:`% @scope_key` - limits the search to the scope bound to
+     the delimiting key :n:`@scope_key`, such as, for example, :n:`%nat`.
+     This clause may be used only if :n:`@string` contains a notation string.
+     (see Section :ref:`LocalInterpretationRulesForNotations`)
```

Comment:
The point of referring to `smart_qualid` above was to avoid repeating this explanation, but let's do as you wish.

Reply:
This is more of a merge artifact.  I don't recall the original content before your and my changes.  I would need a bit of time to think through the added functionality in `Search`.  Note that the `smart_qualid` description is a bit specific to `Locate`.

A couple thoughts in this tactic description (I didn't review the PR):
- I'm not keen on splitting up the prodns here.  I've been trying to keep all the syntax together for each command/tactic.  I'd like to hear your thoughts on that.
- "Cf." is commonly used only in academic papers AFAIK.  This is the only section that uses it.  How about if we stick to "See ..."?
- Maybe show `search_query ... search_query` using `{+ search_query }`.  Three sets of ellipsis are a bit much.
- Factor out `head | hyp | concl | headhyp | headconcl` instead of repeating
- Use fewer lines to show the `Context | Definition | Example | Coercion | ...`
- Some things in the wording (and missing details/definitions e.g. head position, conclusion, and the mysterious "partial applicative subterm"), but I don't want to have another long discussion in this PR.

I could tweak this and let you respond.  WYDT?

Reply:
This seems incorrect: "For a conjunction at top-level, the surrounding brackets are not required."  The syntax doesn't allow a conjunction of terms without brackets.  It does allow a single `search_item` though.

Reply:
"SubClass" was omitted from `logical_kind.  For Coercion, Instance and Scheme, don't they correspond to items defined by the commands of with the same names rather than a status?

Reply:
Sentence fragment: "The location of the pattern or constant within a term"

Reply:
> For Coercion, Instance and Scheme, don't they correspond to items defined by the commands of with the same names rather than a status?

I am not sure if this is relevant, but you can declare coercions, instances, and schemes without using the commands `Coercion`, `Instance`, and `Scheme`.  Consider:
```coq
Record > unitT := { ttT :> unit }. (* delcares Build_unitT and ttT as coercions *)
Existing Class unit.
Class foo := { bar :> unit }. (* declares bar as an instance *)
Set Boolean Equality Schemes.
Set Decidable Equality Schemes.
Set Nonrecursive Elimination Schemes.
Record Unit := TT {}. (* defines schemes Unit_rect, Unit_ind, Unit_rec, Unit_sind, Unit_beq, Unit_eq_dec *)
```

Reply:
> Note that the `smart_qualid` description is a bit specific to `Locate`.

Why so? On the contrary, it seems to me that it is a very general mechanism that is used consistently throughout Coq. For instance, `Canonical`, `Arguments`, etc. all accept `smart_qualid`.

>I'm not keen on splitting up the prodns here. I've been trying to keep all the syntax together for each command/tactic. I'd like to hear your thoughts on that.

I've found that this splitting made things easier to explain, so I wouldn't be too strict about keeping all syntax together when the syntax is complex and it makes sense to explain it step by step.

>"Cf." is commonly used only in academic papers AFAIK. This is the only section that uses it. How about if we stick to "See ..."?

OK let's stick to "See".

>Maybe show search_query ... search_query using {+ search_query }. Three sets of ellipsis are a bit much.

I've tried to imitate your style here, so I'd be happy to go with whatever you think is most readable.

>Factor out head | hyp | concl | headhyp | headconcl instead of repeating

In fact, I've factored things a lot already. I would need to see the actual proposition to comment on whether I find it better.

>Use fewer lines to show the Context | Definition | Example | Coercion | ...

This won't fit on a single line though, so gathering random subsets on one line might look weird or even confusing.

>Some things in the wording (and missing details/definitions e.g. head position, conclusion, and the mysterious "partial applicative subterm"), but I don't want to have another long discussion in this PR.

Yes, I think some of these terms would deserve to become glossary definitions. For instance, "partial applicative subterm" could be defined in the section on functions after we talk about function application.

>I could tweak this and let you respond. WYDT?

I'd appreciate if it could be the topic of another PR (since it's pretty unrelated to the current one).

> The syntax doesn't allow a conjunction of terms without brackets

Yes, by providing multiple `search_query`.

>I am not sure if this is relevant, but you can declare coercions, instances, and schemes without using the commands Coercion, Instance, and Scheme.

It is indeed relevant.

>"SubClass" was omitted from logical_kind.

You mean omitted from the feature or from the documentation?

Reply:
> > Note that the `smart_qualid` description is a bit specific to `Locate`.

> Why so?

It's in the `Locate` command.  The explanation (below) is, I think, a bit too specific to read well when referenced from `Search`.  Just some adjustment in the wording.  We'll still hyperlink.

```
   If the argument is:

   * :n:`@qualid` - displays the full name of objects that
     end with :n:`@qualid`, thereby showing the module they are defined in.
   * :n:`@string {? "%" @ident }` - displays the definition of a notation.  :n:`@string`
     can be a single token in the notation such as "`->`" or a pattern that matches the
     notation.  See :ref:`locating-notations`.
```

> I'd appreciate if it could be the topic of another PR (since it's pretty unrelated to the current one).

I already did it in this PR.  How about taking an initial look first, then I'll create another PR?

> >"SubClass" was omitted from logical_kind.

> You mean omitted from the feature or from the documentation?

Omitted from the grammar due to the editing (`SubClass` was removed from `def_token`).  I made it appear here.

Reply:
> >Use fewer lines to show the Context | Definition | Example | Coercion | ...

> This won't fit on a single line though, so gathering random subsets on one line might look weird or even confusing.

FWIW, the order that was there seemed somewhat random as well:

```
      .. prodn::
         logical_kind ::= @thm_token
         | @assumption_token
         | Context
         | Definition
         | Example
         | Coercion
         | Instance
         | Scheme
         | Canonical
         | Field
         | Method
         | Primitive
```

Reply:
>I already did it in this PR. How about taking an initial look first, then I'll create another PR?

If you already did it, then no need to split it.

>Omitted from the grammar due to the editing (SubClass was removed from def_token). I made it appear here.

OK thanks! That was an accidental removal.

>FWIW, the order that was there seemed somewhat random as well

Keywords are on top, and statuses are below, but other than this, yes, there is no particular order to this list.

Reply:
>> Omitted from the grammar due to the editing (SubClass was removed from def_token). I made it appear here.

> OK thanks! That was an accidental removal.

FWIW, I intentionally removed SubClass from def_token in a previous PR--not an accident and not by you.  That you didn't notice led to what I would call an inadvertent omission.